pallets/mips/src/lib.rs

```diff
@@ -153,13 +164,17 @@ impl Default for MipsPriority {
 /// Properties of a referendum
 #[derive(Encode, Decode, Clone, PartialEq, Eq)]
 #[cfg_attr(feature = "std", derive(Debug))]
-pub struct PolymeshReferendumInfo<Hash: Parameter> {
+pub struct PolymeshReferendum<BlockNumber: Default, Proposal> {
     /// The proposal's unique index.
     pub index: MipsIndex,
     /// Priority.
     pub priority: MipsPriority,
+    /// Current state of this MIP.
+    pub state: MipsState,
+    /// Enactment period.
+    pub enactment_period: BlockNumber,
```

Comment:
Since this value is not valid for a `Ratified` state, should it be an option rather than using 0?

Reply:
In this case, `enactment_period` value is managed internally, and we don't need to differentiate both states (`None` & `Some(0)` are invalid values).